src/js/utils/url.js

```diff
@@ -0,0 +1,89 @@
+import document from 'global/document';
```

Comment:
we should evaluate whether the built-in url module gives us everything we care about.